### PR TITLE
Minor test improvements and fixes.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Monkey-patch Reddwarf Client to log every time we create, delete, etc in the tests.
- Fixed WaitForGuestInstallationToFinish to work black box style.
- Fixed copied LoopingCall class.
- Drop RD db before creating it.
- Added command to log into RD MySQL db.
- Changed Redstack to use MySQL instead of Sqlite.

### CODE REVIEW

File: report/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+report.log
```
Comment (hub-cap) [inline report/.gitignore:1]: how come u didnt add a report/report.log in the projects gitignore? was this just so the empty folder would be in the repo?

File: scripts/redstack [hunk 2]
```diff
@@ -206,10 +206,10 @@ install_packages_1() {
...
 add_to_keystone_conf() {
     echo '
-    catalog.RegionOne.reddwarf.publicURL = http://localhost:8779/v0.1/$(tenant_id)s
-    catalog.RegionOne.reddwarf.adminURL = http://localhost:8779/v0.1/$(tenant_id)s
-    catalog.RegionOne.reddwarf.internalURL = http://localhost:8779/v0.1/$(tenant_id)s
-    catalog.RegionOne.reddwarf.name = 'Reddwarf'
+catalog.RegionOne.reddwarf.publicURL = http://localhost:8779/v0.1/$(tenant_id)s
+catalog.RegionOne.reddwarf.adminURL = http://localhost:8779/v0.1/$(tenant_id)s
+catalog.RegionOne.reddwarf.internalURL = http://localhost:8779/v0.1/$(tenant_id)s
+catalog.RegionOne.reddwarf.name = 'Reddwarf'
```
Comment (hub-cap) [inline scripts/redstack:212]: Quotes?

File: report/.gitignore (hunk 1 above)
Comment (TimSimpsonR) [inline report/.gitignore:1]: Yes.

File: scripts/redstack (hunk 2 above)
Comment (TimSimpsonR) [inline scripts/redstack:212]: I know, its strange. It was like that before and working so I didn't want to change it, not yet at least.

File: tests/integration/int_tests.py [hunk 3]
```diff
@@ -152,6 +152,9 @@ def _clean_up():
     for path in sys.path:
         report.log("\t%s" % path)
 
+    from tests.util.client import monkey_patch_reddwarf_client
+    monkey_patch_reddwarf_client()
```
Comment (hub-cap) [inline tests/integration/int_tests.py:156]: can u splain this? maybe in a comment in the code? seems dubious w/o no explanation

**Comment by hub-cap** [comment]
other than my minor nits +1

File: tests/integration/int_tests.py (hunk 3 above)
Comment (TimSimpsonR) [inline tests/integration/int_tests.py:156]: I commented the actual method, should I do it at the call site too?

Basically it changes RDC at runtime so when you do create, delete, reboot, or resize it writes this to report.log. It's another to see what state the instance is in when the tests are over.

Comment (hub-cap) [inline tests/integration/int_tests.py:156]: Thats fine, i see it now.

**Comment by hub-cap** [comment]
+1

File: scripts/redstack [hunk 4]
```diff
@@ -784,6 +792,7 @@ print_usage() {
           rd-client       - Runs Reddwarf client with admin user.
           wipe-logs       - Resets all log files.
           nova-delete     - Deletes a nova instance.
+          rd-sql          - Opens the Reddwarf MySQL database.
```
Comment (cp16net) [inline scripts/redstack:795]: 1337

File: tests/integration/tests/util/client.py [hunk 5]
```diff
@@ -30,6 +30,42 @@
...
+
+
+def monkey_patch_reddwarf_client():
+    """Make some methods in RDC add to the test report."""
+    global CALLED_MONKEY_PATCH
+    if CALLED_MONKEY_PATCH:
+        raise RuntimeError("monkey_patch_reddwarf_client called twice!")
+    CALLED_MONKEY_PATCH = True
+    from reddwarfclient import instances
+    add_report_event_to(instances.Instances, "create")
```
Comment (cp16net) [inline tests/integration/tests/util/client.py:62]: yikes

File: tests/integration/tests/util/client.py [hunk 6]
```diff
@@ -42,6 +78,7 @@ def __init__(self, real_client):
         """Accepts a normal client."""
         self.real_client = real_client
 
+
```
Comment (cp16net) [inline tests/integration/tests/util/client.py:81]: is this pep8?

**Pull request merged**